cmd/private_network_list.go

```diff
@@ -50,50 +48,49 @@ func (c *privateNetworkListCmd) cmdPreRun(cmd *cobra.Command, args []string) err
 }
 
 func (c *privateNetworkListCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	var zones []string
+	client := globalstate.EgoscaleV3Client
+	ctx := gContext
+
+	resp, err := client.ListZones(ctx)
+	if err != nil {
+		return err
+	}
+	zones := resp.Zones
 
 	if c.Zone != "" {
-		zones = []string{c.Zone}
-	} else {
-		zones = utils.AllZones
+		endpoint, err := client.GetZoneAPIEndpoint(ctx, c.Zone)
+		if err != nil {
+			return err
+		}
+
+		zones = []v3.Zone{{APIEndpoint: endpoint}}
 	}
 
 	out := make(privateNetworkListOutput, 0)
-	res := make(chan privateNetworkListItemOutput)
-	done := make(chan struct{})
 
-	go func() {
-		for nlb := range res {
-			out = append(out, nlb)
-		}
-		done <- struct{}{}
-	}()
-	err := utils.ForEachZone(zones, func(zone string) error {
-		ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, zone))
+	var responseError error
+	var errorZones []string
 
-		list, err := globalstate.EgoscaleClient.ListPrivateNetworks(ctx, zone)
-		if err != nil {
-			return fmt.Errorf("unable to list Private Networks in zone %s: %w", zone, err)
-		}
+	for _, zone := range zones {
+		c := client.WithEndpoint(zone.APIEndpoint)
 
-		for _, p := range list {
-			res <- privateNetworkListItemOutput{
-				ID:   *p.ID,
-				Name: *p.Name,
-				Zone: zone,
+		resp, err := c.ListPrivateNetworks(ctx)
+		if err != nil {
+			responseError = err
+			errorZones = append(errorZones, string(zone.Name))
```

Comment:
We are using the package `multierror` already in the CLI: 
e.g: https://github.com/exoscale/cli/blob/master/utils/utils.go#L12
why not using it.

Reply:
https://github.com/exoscale/cli/pull/644/commits/57aa7c6c3b06668870a94c4551837b8b128371f9#diff-0df82f8689d3e8d881be0653ab4e9f8221c5835ec95a14930d11669e9c7f07efR241

i decided to do this...is it ok